### DISCUSSION TITLE
@ConfigMapping throws ExceptionInInitializerError

### DISCUSSION THREAD

**Discussion by ichrist-gr** [quarkusio/quarkus#24529] [Q&A] [answered]
I am following exactly the quarkus documentation: [https://quarkus.io/guides/config-mappings](url)
and i want to use my configuration interface in Utils class, which is not a CDI and it contains only
static methods. Therefore, i have the following code

SharingConfigs Interface:
`@StaticInitSafe
@ConfigMapping(prefix = "sharing")
public interface SharingConfigs {

  @WithName("amount")
  @WithDefault("amount")
  String amount();
}`

Utils class:
`public class MappingUtils {
  static SmallRyeConfig unwrap = ConfigProvider.getConfig().unwrap(SmallRyeConfig.class);
  static SharingConfigs sharingConfigs = unwrap.getConfigMapping(SharingConfigs.class);
  static RestConfigs restConfigs = unwrap.getConfigMapping(RestConfigs.class);
}`

As you well noticed, in MappingUtils, i'm trying to access a second configuration interface (RestConfigs), which is similar to SharingConfigs, but there is no problem, because a have already injected RestConfigs in a CDI context, which is  annotated with @ApplicationScoped. So, RestConfigs is working as expected.

**The problem apparently appears**, when i do not inject an interface annotated with @ConfigMapping in a CDI context and i'm trying to access it through the SmallRye API.

What is the problem? 
java.lang.ExceptionInInitializerError
Caused by: java.util.NoSuchElementException: SRCFG00027: Could not find a mapping for SharingConfigs

Any ideas to solve this issue?

**Comment by radcortez** [comment, marked as answer]
Yes, unused mappings are removed from Quarkus. Unfortunately, we cannot detect programmatic cases, so you need to add the annotation `@Unremovable` to the mapping class.

**Reply by ichrist-gr** [reply]
That seems to be the solution to the problem. Thank you very much for the response

**Comment by floresmleonardo** [comment]
The solution explained work, I just like to know if I can do it using a different approach.

I am creating a quarkus extension for a framework, which uses **ConfigMappping**. Inside my QuarkusProcessor I have tried different approaches to mark the **ConfigMapping** as **Unremovable**:

```java
  @BuildStep
  UnremovableBeanBuildItem addUnremovable() {
    return UnremovableBeanBuildItem.beanTypes(MyConfigMapping.class);
  }
```

I also try the following:

```java
 @BuildStep
  AdditionalBeanBuildItem addUnremovable() {
    return AdditionalBeanBuildItem.builder()
        .addBeanClass(MyConfigMapping.class)
        .setUnremovable()
        .build();
  }
```

The reason to try the approaches is to separate quarkus dependencies libraries as much as possible from the framework. Both approaches produce the same error.

For now, I am using the annotation on the ConfigMapping class, and works as expected. Thank you for the help that you can provide.

**Reply by radcortez** [reply]
Hum, most likely, we are not looking for beans marked as unremovable programmatically to register them with the Configuration system. Do you mind opening up an issue so we can look at it? Thank you!

**Reply by floresmleonardo** [reply]
Sure. Thank you!
